src/modules/emote_menu/stores/emote-storage.js

```diff
@@ -0,0 +1,119 @@
+import storage from '../../../storage.js';
+
+const HOUR = 60 * 60 * 60 * 1000;
+const DAY = 24 * HOUR;
+
+const MAX_TIMESTAMPS = 100;
+
+function timestampToScore(timestamp) {
+  const diff = Date.now() - timestamp;
+
+  switch (true) {
+    case diff < 4 * HOUR:
+      return 100;
+    case diff < DAY:
+      return 80;
+    case diff < 3 * DAY:
+      return 60;
+    case diff < 7 * DAY:
+      return 40;
+    case diff < 30 * DAY:
+      return 20;
+    case diff < 90 * DAY:
+      return 10;
+    default:
+      return 0;
+  }
+}
+
+function calcScore(totalUses, recentUses) {
+  const frecency = recentUses.reduce((a, b) => a + timestampToScore(b), 0);
+  return Math.floor((totalUses * frecency) / recentUses.length);
+}
+
+class EmoteStorage {
+  constructor() {
+    this.emoteStore = storage.get('emotes');
+    this.ids = [];
+
+    if (this.emoteStore == null) {
+      this.emoteStore = {
+        usageHistory: {},
+        favorites: [],
+      };
+    }
+
+    this.updateAllScores();
+    this.updateIds();
+  }
+
+  updateAllScores() {
+    for (const [id, {totalUses, recentUses}] of Object.entries(this.emoteStore.usageHistory)) {
+      this.emoteStore.usageHistory[id].score = calcScore(totalUses, recentUses);
+    }
+  }
+
+  toggleFavorite(emote) {
```

Comment:
maybe `setFavorite` and accept a deliberate value instead? otherwise toggling favorites across windows could result in unexpected behavior?